File: src/main/java/com/ltammie/problems/blind75/TwoSum.java
package com.ltammie.problems.blind75;

import java.util.HashMap;
import java.util.Map;

/**
 * @No 1
 */
public class TwoSum {

    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (map.containsKey(complement) && map.get(complement) != i) {
                return new int[]{map.get(complement), i};
            }
            map.put(nums[i], i);
        }

        return new int[]{0, 0};
    }
}

File: src/main/kotlin/com/ltammie/problems/solutions/TwoSum.kt
package com.ltammie.problems.solutions

/**
 * @theme Top Interview Questions
 * @No 1
 */
fun twoSum(nums: IntArray, target: Int): IntArray {
    val map: MutableMap<Int, Int> = HashMap()
    for (i in nums.indices) {
        val complement = target - nums[i]
        if (map.containsKey(complement)) {
            return intArrayOf(i, map[complement]!!)
        }
        map[nums[i]] = i
    }
    return intArrayOf(0, 1)
}